### ISSUE TITLE
Autoform for user roles

### ISSUE THREAD

**Issue by nick-lehmann** [Meteor-Community-Packages/meteor-autoform#1391]
I am currently in the process of creating an admin form to create new users (using autoform). To assign specific privileges to the users, I added `alanning:roles` to my project. I used the schema proposed by the documentation of `aldeed:collection2` which includes the following definition for roles:

```
roles: {
    type: Object,
    optional: true,
    blackbox: true
},
```

It makes sense to me that roles are defined as an object in order to support the different groups that can be assigned. Unfortunately, it does not make creating the form very easy. By using `afQuickField` I get a form element that is not useful at all. I would like to have `select-multiple` element which proposes all available roles to the admin.

At the moment, SimpleSchema prints out the debug information that it has sorted out the values that would have affected roles.

My first idea was to set the type of the form to `method` and to specify a `meteormethod`that will take the chosen roles, wrap it into a hash and then insert the user. Since I already have a method to create users (for the register form) this solution seemed handy. Unfortunately, I forgot that the form will only submitted if it passes the validation, which does not happen.

My next idea was to use Autoform hooks. All hooks I tried to use (´onSubmit´, ´formToDoc´ and ´before´) are invoked after ´SimpleSchema.clean´ has filtered out the chosen roles.

All ideas are welcome!

**Comment by Tim-W** [comment]
Have you tried using the 'onSuccess' hook? That worked for me.

**Comment by nick-lehmann** [comment]
If I am not mistaken, this hook is fired when everything worked fine. I tried it and the same issue occurs. 

Maybe I did not make my point clear enough. The modification of the field/data must happen before the validation takes place.

**Comment by aleritty** [comment]
I have a similar problem, I solved part of it using "normal" as form type and insime onSubmit hook doing manually the validation (I used a schema not attached to meteor.users because attaching it seems to break everything in my scenario).
Then I can call my methods to insert the user, adding the roles and return false (so the form wil not try to insert).

But the form will not render correctly if I pass a doc to it (for updating an existing user). I choose to show the roles as checkboxes, but also select-multiple didn't work.

**Comment by U54** [comment]
I'm trying to achieve the same type of thing but having very similar problems, did you manage to get it working? If anyone has it'd be really helpful maybe to make a simple gist of it or share how, seem to be running into a bunch of issues with this...

**Comment by nick-lehmann** [comment]
As it seems to be a common issue, maybe @aldeed has an idea. Do you have any hint?

**Comment by aleritty** [comment]
I solved simply NOT attaching a schema to Meteor.users.
Actually I use a schema with just the profile fields, email and password, then I use autoform with "normal" type, my own functions and insert works.
For Updating I need to fetch the user fields, build an object with the same structure of the schema

I made a gist with my code, maybe can be useful... the code is wrong in some ways I know, but it works. Sorry if some parts are in Italian

https://gist.github.com/aleritty/d025313de556881d0951

Maybe we can attract the meteor authors on this issue, so the can give official indication at aldeed, or give us an official way to do this things (maybe a separate package)
